client/client.go

```diff
@@ -129,27 +128,14 @@ func (c *DatabricksClient) redactedDump(prefix string, body []byte) (res string)
 	}.redactedDump(prefix, body)
 }
 
-// Fully read and close HTTP response body to release HTTP connections.
-//
-// HTTP client's Transport may not reuse HTTP/1.x "keep-alive" TCP connections
-// if the Body is not read to completion and closed.
-//
-// See: https://groups.google.com/g/golang-nuts/c/IoSvPz-rpfc
-func (c *DatabricksClient) drainToEnsureConnectionRelease(r *http.Response) {
-	if r == nil || r.Body == nil {
-		return
-	}
-	defer r.Body.Close()
-	_, err := io.Copy(ioutil.Discard, r.Body)
-	if err != nil {
-		logger.Errorf("failed to drain body: %s", err)
-	}
-}
-
-func (c *DatabricksClient) attempt(ctx context.Context,
-	method, requestURL string, requestBody []byte,
-	visitors ...func(*http.Request) error) func() (*http.Response, *retries.Err) {
-	return func() (*http.Response, *retries.Err) {
+func (c *DatabricksClient) attempt(
+	ctx context.Context,
+	method string,
+	requestURL string,
+	requestBody []byte,
+	visitors ...func(*http.Request) error,
+) func() (*bytes.Buffer, *retries.Err) {
```

Comment:
How returning `*bytes.Buffer` would affect our ability to read response headers?

Reply:
It wouldn't allow for that, but we don't do this today. If we need to we can add back the `http.Response`. 

I removed it because we can only use a single return argument from `retries.Poll` and would need to wrap a tuple with `http.Response` and `bytes.Buffer` for the body otherwise.